src/main/java/com/celements/search/web/WebSearchService.java

```diff
@@ -0,0 +1,119 @@
+package com.celements.search.web;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentNotExistsException;
+import com.celements.model.context.ModelContext;
+import com.celements.rights.access.EAccessLevel;
+import com.celements.rights.access.IRightsAccessFacadeRole;
+import com.celements.search.lucene.ILuceneSearchService;
+import com.celements.search.lucene.LuceneSearchResult;
+import com.celements.search.lucene.query.LuceneQuery;
+import com.celements.search.web.classes.WebSearchConfigClass;
+import com.celements.search.web.packages.WebSearchPackage;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.web.Utils;
+
+@Component
+public class WebSearchService implements IWebSearchService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebSearchService.class);
+
+  @Requirement
+  private IModelAccessFacade modelAccess;
+
+  @Requirement
+  private IRightsAccessFacadeRole rightsAccess;
+
+  @Requirement
+  private ILuceneSearchService luceneSearchService;
+
+  @Requirement
+  private List<WebSearchPackage> webSearchPackages;
+
+  @Requirement
+  private ModelContext context;
+
+  @Override
+  public Set<WebSearchPackage> getAvailablePackages(DocumentReference configDocRef) {
+    Set<WebSearchPackage> searchPackages = new HashSet<>();
+    try {
+      XWikiDocument configDoc = modelAccess.getDocument(configDocRef);
+      searchPackages.addAll(getConfiguredPackages(configDoc));
+      if (searchPackages.isEmpty()) {
+        searchPackages.addAll(getDefaultPackages());
+      }
+      searchPackages.addAll(getRequiredPackages(configDocRef));
+    } catch (DocumentNotExistsException exp) {
+      LOGGER.error("Failed to load configDoc '{}'", configDocRef, exp);
```

Comment:
I think error level is too much, better warn and add default packages to return value as a reasonable fallback